R/zchunk_L102.ghg_en_USA_S_T_Y.R

```diff
@@ -31,46 +31,77 @@ module_emissions_L102.ghg_en_USA_S_T_Y_DISABLED <- function(command, ...) {
+      select(-CO2) %>% # non-CO2 only
+      group_by(sector, fuel) %>%
+      summarize_if(is.numeric, sum, na.rm = FALSE) %>% # sum by sector and fuel
+      filter(!is.na(sector), !is.na(fuel)) %>% # delete NA sectors and fuel
+      mutate_all( funs( replace(., is.na(.), 1))) %>% # placeholder values for existing sectors that has no value.
+      # NOTE: THIS IS A HACK. EPA DOESN'T HAVE EMISSIONS IN SOME SECTORS, WHEN EDGAR DOES. THIS WILL MAKE EMISSIONS PROPORTIONAL TO FUEL USE. NOT SURE IF THIS IS THE BEST STRATEGY.
+      mutate_if(is.numeric, funs(. * CONV_GG_TG)) ->  # Convert to Tg
+      L102.ghg_tg_USA_en_Sepa_F_2005 # GHG balance in 2005
+
+    if(OLD_DATA_SYSTEM_BEHAVIOR) {
+    # incorrect fuel name for transport from input mapping.
+    # organize energy balances in USA 2005
+        L101.in_EJ_R_en_Si_F_Yh %>% # start from energy balances
+        filter(GCAM_region_ID == gcam.USA_CODE) %>% # USA region only
+        select(-GCAM_region_ID) %>%
+        gather(variable, energy, -sector, -fuel, -technology) %>%
+        filter(variable == "X2005") %>%  # 2005 data only
+        left_join_keep_first_only( select(GCAM_sector_tech, sector, fuel, EPA_agg_sector, EPA_agg_fuel_ghg),
+                    by = c( "sector", "fuel" )) %>% # assign aggregate sector and fuel names
+        group_by(EPA_agg_sector, EPA_agg_fuel_ghg) %>%
+        summarize_if(is.numeric, sum) -> # sum by aggregate sector and fuel
+        L102.in_EJ_USA_en_Sepa_F_2005 # energy balance in 2005
+
+    } else {
+      # fool proof solution is to use both fuel and technology for categorization.
+      # organize energy balances in USA 2005
+      L101.in_EJ_R_en_Si_F_Yh %>% # start from energy balances
+        filter(GCAM_region_ID == gcam.USA_CODE) %>% # USA region only
+        select(-GCAM_region_ID) %>%
+        gather(variable, energy, -sector, -fuel, -technology) %>%
+        filter(variable == "X2005") %>%  # 2005 data only
+        left_join( select(GCAM_sector_tech, sector, fuel, technology, EPA_agg_sector, EPA_agg_fuel_ghg),
+                   by = c( "sector", "fuel", "technology" )) %>% # assign aggregate sector and fuel names
+        group_by(EPA_agg_sector, EPA_agg_fuel_ghg) %>%
+        summarize_if(is.numeric, sum) -> # sum by aggregate sector and fuel
+        L102.in_EJ_USA_en_Sepa_F_2005 # energy balance in 2005
+    }
+
+    # combine emissions and energy to get emission factors
+    L102.ghg_tg_USA_en_Sepa_F_2005 %>%
+      left_join(L102.in_EJ_USA_en_Sepa_F_2005, by = c( "sector" = "EPA_agg_sector", "fuel" = "EPA_agg_fuel_ghg" )) %>% # energy data and emission data joined
+      mutate(ch4_em_factor = CH4/energy) %>% # emission factor calculated
+      mutate(n2o_em_factor = N2O/energy) %>% # emission factor calculated
+      select(-CH4, -N2O, -energy) %>% # delete orignal data
+      arrange(fuel) %>%
+      mutate_all( funs( replace(., is.na(.), 0))) %>% # zero out NA
```

Comment:
This is a pretty baroque way to do this--hard to see what's going on. Worth reshaping/changing for clarity? Also remove GCAM-style spacing